server/SaveLoad.cpp

```diff
@@ -172,6 +172,7 @@ int SaveGame(const std::string& filename, const ServerSaveGameData& server_save_
         pos_before_writing = ofs.tellp();
 
         bool save_completed_as_xml = false;
+        galaxy_setup_data.EncodingEmpire() = ALL_EMPIRES;
```

Comment:
Rather than passing the `GalaxySetupData` by non-const reference and changing the interfaces everywhere, I'd pass it by value and modify the local, or by const reference and make a local copy.

Reply:
Fixed